### PR TITLE
Enable search engines summary

### LINKED ISSUE #62: Add search engine score summaries
Based on idXMLs.

Histograms per search engine:
 - one series per file
 - always targets (dark color) and decoys (light color)
 - use e.g. specEvalue for MSGF and xcorr for comet

Histogram per file (e.g. dropdown menu) or histogram after merging
 - one series per search engine, after Percolator/IDPEP
 - always targets (dark color) and decoys (light color)
 - use PEP or PP as score

Either: Barplot on number of matching search engine IDs per PSM
Or: Scatterplot of number of matching search engine IDs per PSM versus the best PEP/PP score

### PR DESCRIPTION
**Pull request by WangHong007**
fixes #62 
New example is here: [report](https://github.com/WangHong007/Data-Folder/blob/511dd8a64d9241f8fc87e744a6f48e44c8471f6f/pmultiqc/multiqc_report_7.html)

### CODE REVIEW

**Review by ypriverol — APPROVED** [review]

**Comment by jpfeuffer** [comment]
Did you see my comment in the issue about the consensus plot?

**Comment by ypriverol** [comment]
@WangHong007 can you load an example for `spectral counting` quantms analysis. I would like to see if the new changes proposed by @daichengxin are also seen with the new plots of identifications.

**Comment by ypriverol** [comment]
> Did you see my comment in the issue about the consensus plot?

@WangHong007, @jpfeuffer is talking about this comment https://github.com/bigbio/pmultiqc/issues/62#issuecomment-1195498670

**Comment by WangHong007** [comment]
@jpfeuffer You mean count how many search engines identify each PSM. In this case, the number of bars per file is the number of search engines (1,2 or future 3,4...), single search engine will also be plotted.
Also, this part does not need idXMLs after `consensus`, but needs to process idXMLs from multiple engines.

**Comment by WangHong007** [comment]
@ypriverol Got it.

**Comment by jpfeuffer** [comment]
As mentioned before, the "consensus" Metavalue should show you that

**Comment by jpfeuffer** [comment]
https://github.com/OpenMS/OpenMS/blob/develop/src/tests/topp/ConsensusID_4_output.idXML#L19

**Comment by WangHong007** [comment]
Just counting the frequency of different `consensus_supports`, right?

**Comment by WangHong007** [comment]
New example is here: [report](https://github.com/WangHong007/Data-Folder/blob/00f1b8ef98502c22f37080f31d9914e70217bcb5/pmultiqc/multiqc_report_13.html)

**Comment by jpfeuffer** [comment]
Great! I think I made a mistake :D Looks like `support * nr_engines` does not work. Maybe call it "Consensus across search engines" and just put `support` there. I can then explain later in the help text what it means.

Otherwise the PR is good to go. Maybe in a later PR we can find a way to limit the histograms to the actual max. of the data but you probably need to parse all files of a search engine first, then create the empty histogram and then fill it.

File: pmultiqc/modules/quantms/quantms.py [hunk 1]
```diff
@@ -750,6 +755,109 @@ def draw_delta_mass(self):
                     ''',
             plot=line_html
         )
+    
+    def draw_search_engine(self):
+        # Add a report section with multiple histograms
+        self.add_section(
+            name="Summary of Search Engine",
```
Comment (jpfeuffer) [inline pmultiqc/modules/quantms/quantms.py:762]:
```suggestion
            name="Summary of Search Engine Scores",
```

**Comment by WangHong007** [comment]
The final report here: [pmultiqc_#66](https://github.com/WangHong007/Data-Folder/blob/11cb95bf59fd3b319dd1ebd9ee74d39200696cb6/pmultiqc/multiqc_report_1.html)

**Comment by WangHong007** [comment]
Final report here: [report_#60_#66](https://github.com/WangHong007/Data-Folder/blob/aaf23e51390d662eed008174a14346e5cf510da5/pmultiqc/multiqc_report_2.html)

File: pmultiqc/modules/quantms/quantms.py [hunk 2]
```diff
@@ -750,6 +755,109 @@ def draw_delta_mass(self):
...
+                'stacking': True,
+                'height': 256,
+                'tt_percentages': True, 
+                'tt_decimals': 0,
+                'data_labels': self.search_engine['data_label']['consensus_label'],
+            }
+            consensus_bar_html = bargraph.plot(list(self.search_engine['consensus_support'].values()), PEP_cats, consensus_pconfig)
+            
+            self.add_section(
+                description='''#### Summary of consensus PSMs
```
Comment (jpfeuffer) [inline pmultiqc/modules/quantms/quantms.py:851]:
```suggestion
                description='''#### Summary of consensus support for PSMs
                Consensus support is a measure of agreement between search engines. Every peptide
                sequence in the analysis has been identified by at least one search run. The consensus
                support defines which fraction (between 0 and 1) of the remaining search runs "supported"
                a peptide identification that was kept. The meaning of "support" differs slightly between
                algorithms: For best, worst, average and rank, each search run supports peptides that it has
                also identified among its top considered_hits candidates. So the "consensus support" simply
                gives the fraction of additional search engines that have identified a peptide. (For example, if
                there are three search runs, peptides identified by two of them will have a "support" of 0.5.)
                For the similarity-based algorithms PEPMatrix and PEPIons, the "support" for a peptide is the
                average similarity of the most-similar peptide from each (other) search run.
```

**Review by jpfeuffer — APPROVED** [review]

**Pull request merged**